ionic_user_interface/src/views/ResetPassword.vue

```diff
@@ -0,0 +1,309 @@
+                      <password-meter @score="onPasswordScore" :password="passwordText" />
+                    </ion-col>
+                    <ion-col size="auto" class="password-strength">
+                      {{ passwordStrength }}
+                    </ion-col>
+                  </ion-row>
+                </div>
+                <ion-item class="rounded margin">
+                  <ion-label position="stacked">Retype new password</ion-label>
+                  <ion-input
+                    @keyup.enter="clickResetPasswordButton"
+                    v-model="confirmPasswordText"
+                    name="confirmpassword"
+                    type="password"
+                    required
+                  />
+                </ion-item>
+                <ion-button
+                  id="resetPasswordButton"
+                  class="reset-password-button"
+                  size="medium"
+                  type="submit"
+                  expand="block"
+                >
+                  Reset Password
+                </ion-button>
+                <ion-button
+                  id="resendButton"
+                  class="resend-button"
+                  size="medium"
+                  expand="block"
+                  fill="clear"
+                  @click="resendPasswordResetCode"
+                >
+                  Resend Code
+                </ion-button>
+              </form>
+            </div>
+          </ion-col>
+        </ion-row>
+      </ion-grid>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script lang="ts">
+import {
+  IonButton,
+  IonCol,
+  IonContent,
+  IonGrid,
+  IonInput,
+  IonItem,
+  IonLabel,
+  IonPage,
+  IonRow,
+  onIonViewDidLeave,
+  toastController,
+} from '@ionic/vue';
+import { defineComponent, inject, ref, Ref, watch, onMounted } from 'vue';
+import axios from 'axios';
+import PasswordMeter from 'vue-simple-password-meter';
+import MessageBox from '@/components/MessageBox.vue';
+import { useRouter } from 'vue-router';
+import { throttle } from 'lodash';
+
+export default defineComponent({
+  name: 'ResetPassword',
+  components: {
+    MessageBox,
+    IonButton,
+    IonCol,
+    IonContent,
+    IonGrid,
+    IonInput,
+    IonItem,
+    IonLabel,
+    IonPage,
+    IonRow,
+    PasswordMeter,
+  },
+  setup() {
+    const router = useRouter();
+    const msgBox: Ref<typeof MessageBox | null> = ref(null);
+    const msgBoxColor = ref('danger');
+    const usernameText = ref('');
+    const passwordResetCodeText = ref('');
+    const passwordText = ref('');
+    const confirmPasswordText = ref('');
+    const passwordStrength = ref('');
+    const getUsername: () => Ref<string> = inject('getUsername', () => ref(''));
+
+    onIonViewDidLeave(() => {
+      msgBox.value?.close();
+    });
+
+    onMounted(() => {
+      usernameText.value = getUsername().value;
+      msgBoxColor.value = 'medium';
+      // Email is not displayed explicitly for security
```

Comment:
We should display some semblance of the email though, like most websites do. For e.g. `e***l@gmail.com` or something